src/fw-logs/fw-logs-parser.cpp

```diff
@@ -1,90 +1,215 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
-#include "fw-logs-parser.h"
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include <src/fw-logs/fw-logs-parser.h>
+#include <src/fw-logs/fw-string-formatter.h>
+#include <src/fw-logs/fw-logs-xml-helper.h>
+
+#include <rsutils/string/from.h>
+
 #include <regex>
-#include <sstream>
-#include "fw-string-formatter.h"
-#include "stdint.h"
+#include <fstream>
+#include <iterator>
 
-using namespace std;
 
 namespace librealsense
 {
     namespace fw_logs
     {
-        fw_logs_parser::fw_logs_parser(string xml_content)
-            : _fw_logs_formating_options(xml_content),
-            _last_timestamp(0),
-            _timestamp_factor(0.00001)
+        fw_logs_parser::fw_logs_parser( const std::string & definitions_xml )
+            : _source_to_formatting_options()
         {
-            _fw_logs_formating_options.initialize_from_xml();
+            fw_logs_xml_helper helper;
+            _source_id_to_name = helper.get_listed_sources( definitions_xml );
+            for( auto & source : _source_id_to_name )
+            {
+                std::string path = helper.get_source_parser_file_path( source.first, definitions_xml );
+                std::ifstream f( path.c_str() );
+                if( f.good() )
+                {
+                    std::string xml_contents;
+                    xml_contents.append( std::istreambuf_iterator< char >( f ), std::istreambuf_iterator< char >() );
+                    fw_logs_formatting_options format_options( std::move( xml_contents ) );
+                    _source_to_formatting_options[source.first] = format_options;
+                }
+                else
+                    throw librealsense::invalid_value_exception( rsutils::string::from() << "Can't open file " << path );
+            }
         }
 
+        fw_log_data fw_logs_parser::parse_fw_log( const fw_logs_binary_data * fw_log_msg ) 
```

Comment:
This will hold a single full log in both legacy and new flows?

Reply:
Yes. It uses `structured_binary_data` to put the binary data in a common structure, than parses it to strings.

`fw_logs_parser` base class handles the extended format, `legacy_fw_logs_parser` overloads handle the legacy format.